apps/system/js/app_install_manager.js

```diff
@@ -498,6 +498,12 @@
 
     handleDownloadSuccess: function ai_handleDownloadSuccess(evt) {
       var app = evt.application;
+      var evtDownload = new CustomEvent('downloadsuccess',
+        { bubbles: true,
+          cancelable: false,
+          detail: { application: app } });
+      window.dispatchEvent(evtDownload);
```

Comment:
I don't understand why you need to dispatch this event on |window| since you will add your event listener on the app object itself in |recordInstall|